external-crates/move/crates/move-binary-format/src/compatibility.rs

```diff
@@ -301,34 +319,9 @@ impl Compatibility {
         let old_friend_module_ids: BTreeSet<_> = old_module.friends.iter().cloned().collect();
         let new_friend_module_ids: BTreeSet<_> = new_module.friends.iter().cloned().collect();
         if !old_friend_module_ids.is_subset(&new_friend_module_ids) {
-            friend_linking = false;
+            friend_linking!("removed friend module declaration");
         }
 
-        if self.check_datatype_and_pub_function_linking && !datatype_and_function_linking {
-            return Err(PartialVMError::new(
-                StatusCode::BACKWARD_INCOMPATIBLE_MODULE_UPDATE,
-            ));
-        }
-        if self.check_datatype_layout && !datatype_layout {
-            return Err(PartialVMError::new(
-                StatusCode::BACKWARD_INCOMPATIBLE_MODULE_UPDATE,
-            ));
-        }
-        if self.check_friend_linking && !friend_linking {
-            return Err(PartialVMError::new(
-                StatusCode::BACKWARD_INCOMPATIBLE_MODULE_UPDATE,
-            ));
-        }
-        if self.check_private_entry_linking && !entry_linking {
-            return Err(PartialVMError::new(
-                StatusCode::BACKWARD_INCOMPATIBLE_MODULE_UPDATE,
-            ));
-        }
-        if self.disallow_new_variants && !no_new_variants {
-            return Err(PartialVMError::new(
-                StatusCode::BACKWARD_INCOMPATIBLE_MODULE_UPDATE,
-            ));
-        }
```

Comment:
Since the tests that test this error are auto-generated anyway and expect a specific error (including the correct message), I don't think it makes sense to put any effort into making the order deterministic somehow.

I also don't think that listing all the errors will be beneficial (we are not building an IDE that would benefit from seeing all errors at the same time. We just want to check if we can update or not and should fail as early as possible without wasting any additional checks once this becomes clear, no?